src/plugins/sqlite.js

```diff
@@ -0,0 +1,126 @@
+"use strict";
+
+const path = require("path");
+const fsextra = require("fs-extra");
+const sqlite3 = require("sqlite3");
+const Helper = require("../helper");
+const Msg = require("../models/msg");
+
+const currentSchemaVersion = 1;
+
+const schema = [
+	// Schema version #1
+	"CREATE TABLE IF NOT EXISTS options (name TEXT, value TEXT, CONSTRAINT name_unique UNIQUE (name))",
+	"CREATE TABLE IF NOT EXISTS messages (network TEXT, channel TEXT, time INTEGER, type TEXT, msg TEXT)",
+	"CREATE INDEX IF NOT EXISTS network_channel ON messages (network, channel)",
+	"CREATE INDEX IF NOT EXISTS time ON messages (time)",
+];
+
+class MessageStorage {
+	constructor(client) {
+		this.isEnabled = client.config.log && Helper.config.messageStorage === "sqlite";
```

Comment:
Could we just ... not require this if the user doesn't specify sqlite?

Reply:
Yes enabling needs to be moved outside.

Reply:
We can probably leave this till we allow plugins if you want though.